feat: implement webtoon-style viewer for tall images

omnessssssssss opened the pull request:
- Add fitWidthForTallImages parameter to BooruImage widget
- Always use BoxFit.fitWidth when fitWidthForTallImages flag is enabled
- Remove aspect ratio constraints for tall images to allow proper fit-width
- Detect very tall images (aspect ratio > 2.0) in post details
- Use SingleChildScrollView for tall images instead of InteractiveViewer
- Add proper status bar padding for full-screen tall image viewing
- Preserve tap-to-toggle overlay functionality for tall images
- Allow natural vertical scrolling instead of pan-to-info gesture
- Update OriginalImagePage and PostDetailsImage to use new fit mode

This change allows tall images to fit the screen width and be scrollable vertically, providing a webtoon-style viewing experience for long images.

Fix #524

omnessssssssss committed:
feat: implement webtoon-style viewer for tall images

- Add fitWidthForTallImages parameter to BooruImage widget
- Always use BoxFit.fitWidth when fitWidthForTallImages flag is enabled
- Remove aspect ratio constraints for tall images to allow proper fit-width
- Detect very tall images (aspect ratio > 2.0) in post details
- Use SingleChildScrollView for tall images instead of InteractiveViewer
- Add proper status bar padding for full-screen tall image viewing
- Preserve tap-to-toggle overlay functionality for tall images
- Allow natural vertical scrolling instead of pan-to-info gesture
- Update OriginalImagePage and PostDetailsImage to use new fit mode

This change allows tall images to fit the screen width and be scrollable
vertically, providing a webtoon-style viewing experience for long images.

omnessssssssss committed:
Remove unused variables and fix linter warnings

khoadng commented on lib/core/posts/details/src/widgets/post_details_item.dart:
```diff
@@ -117,75 +118,138 @@ class PostDetailsItem<T extends Post> extends ConsumerWidget {
-                      : const SizedBox.shrink(),
-                ),
-            ],
+          child: _buildPostContent(
```
Don't create new method, just keep it inline.

khoadng commented:
1. Swiping up to pull up the details sheet no longer works. This is somewhat okay since version 4.2 introduced a button to open and close the sheet. However, the app now has a new TikTok-style swipe feature, which this change will break.

2. The way to determine when to use the tall image mode is pretty arbitrary, I don't like it.

I have been thinking about adding a dedicated mode in the settings for this kind of usage. The next release will introduce a settings to change between a gallery browsing mode and a tiktok browsing mode. Maybe adding a comic mode would be better

omnessssssssss committed:
refactor: inline post content method and fix generic type issues

 - Remove _buildPostContent method and inline content as requested in PR review
 - Fix PostDetailsImage generic type parameter to properly handle T extends Post
 - Fix state class to maintain generic type information

omnessssssssss committed:
fix: restore swipe-up for post info on tall images when UI is visible

- Add swipe-up gesture detection for tall images when overlay UI is showing
- Disable scrolling when UI is visible to allow swipe gesture to work
- Enable normal scrolling when UI is hidden for viewing tall images

khoadng commented:
Related #524

omnessssssssss committed:
chore: merge upstream master and resolve conflicts

- Resolved import conflict in post_details_item.dart
- Regenerated translation files to include new keys from upstream
- Merged changes for backup/restore, post grid, and settings features

omnessssssssss committed:
Merge branch 'master' into tall-images-viewer

omnessssssssss committed:
feat: add tall media viewer with scrollable support for tall images

Implements intelligent tall media detection and specialized scrolling behavior for very tall images/comics. Adds settings-driven classification system to determine when content should use scrollable viewer versus standard interactive viewer.

(Trying Codex instead of Claude this time :P)

omnessssssssss commented:
@khoadng Some more AI changes 😆 

Tested on virtual device on a couple of images and it does work for both horizontal and vertical scroll modes

khoadng commented on lib/core/settings/src/types/settings.dart:
```diff
@@ -554,6 +567,121 @@ class ImageViewerSettings extends Equatable {
+    required this.minScrollExtentPx,
+  });
+
+  const TallMediaSettings.defaults()
```
This is too much for an user settings. Should have just create a new value in PostDetailsSwipeMode, maybe comic or webtoon.

khoadng commented on lib/core/posts/details/src/utils/tall_media_classifier.dart:
```diff
@@ -0,0 +1,137 @@
+  bool get hasScrollableExtent => scrollExtent > 0;
+}
+
+class TallMediaClassifier {
```
No need for class, just a function is enough.

khoadng commented:
Add some quick feedback, will check the rest later

omnessssssssss commented:
I will resolve the comments soon however I think you still have a misunderstanding of what it is I want. Hopefully this can help
<img width="5900" height="2973" alt="Untitled-2025-09-17-2047" src="https://github.com/user-attachments/assets/aa86503e-254f-471b-ab67-ae185adb8215" />

omnessssssssss committed:
refactor(tall-media): simplify settings and convert classifier to function

- Reduce TallMediaSettings from 10 complex parameters to single enabled toggle
- Convert TallMediaClassifier class to classifyTallMedia function
- Move threshold values to static constants for consistent behavior
- Add feature disable capability via enabled setting
- Update all usage sites and tests for new function-based API

khoadng commented on lib/core/posts/details/src/utils/tall_media_classifier.dart:
```diff
@@ -0,0 +1,129 @@
+  bool get hasScrollableExtent => scrollExtent > 0;
+}
+
+TallMediaDisposition classifyTallMedia({
```
- this should be a factory method in TallMediaDisposition, rename it to fromViewport. Move all calculation to this method
- settings.enabled check, it should be handle from the UI

khoadng commented on lib/core/settings/src/types/settings.dart:
```diff
@@ -554,6 +567,45 @@ class ImageViewerSettings extends Equatable {
+  @override
+  List<Object> get props => [enabled];
+
+  static const double aspectRatioThreshold = 2.15;
```
move these to TallMediaConfigs, pass the config as a option parameter to the above mentioned fromViewport method.

khoadng commented on lib/core/posts/details/src/widgets/post_details_image.dart:
```diff
@@ -117,10 +121,11 @@ class _PostDetailsImageState extends ConsumerState<PostDetailsImage> {
       placeholderUrl: placeholderImageUrl,
       aspectRatio: post.aspectRatio,
-      forceCover: post.aspectRatio != null,
+      forceCover: false, // Never force cover when we want fit width
```
This should be conditional based on fitWidthForTallImages

khoadng commented on lib/core/posts/details_pageview/src/post_details_page_view_controller.dart:
```diff
@@ -14,6 +14,12 @@ import 'auto_slide_mixin.dart';
 import 'post_details_page_view.dart';
 
+enum SwipeLockReason {
+  legacy,
```
What is legacy?
